geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/KryoLazyDensityFilter.java

```diff
@@ -0,0 +1,359 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.hbase.filters;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.exceptions.DeserializationException;
+import org.apache.hadoop.hbase.filter.FilterBase;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.geotools.factory.Hints;
+import org.geotools.filter.text.ecql.ECQL;
+import org.geotools.util.Converters;
+import org.locationtech.geomesa.features.ScalaSimpleFeature;
+import org.locationtech.geomesa.features.interop.SerializationOptions;
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer;
+import org.locationtech.geomesa.index.conf.QueryHints;
+import org.locationtech.geomesa.utils.geometry.Geometry;
+import org.locationtech.geomesa.utils.geotools.GridSnap;
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+import java.io.*;
+import java.util.AbstractMap;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opengis.feature.type.GeometryDescriptor;
+import org.opengis.filter.expression.Expression;
+import scala.Tuple2;
+
+public class KryoLazyDensityFilter extends FilterBase {
+
+    private static String sftString;
```

Comment:
So is there only a single ```KryoLazyDensityFilter``` for the entire hbase server JVM? I need to clearly look over the API but just wanted to make sure this isn't initialized simultaneously by 2 separate scans in HBase.